js/sprite/Sprite.js

```diff
@@ -0,0 +1,238 @@
+import {br, numToSpace, removeTimestampDelimiters, reverseTimestamps} from "../stringManipulation.js";
+
+/**
+ * a class to standardize sprite creation and manipulation
+ * @author Michael Crockett
+ */
+
+export default class Sprite {
+    static receiptWidth = 40;
+    /**
+     * The characters that make up the sprite.
+     * Spaces should be non-breaking (represented by `&nbsp;`) and lines are separate elements in the array.
+     * @type {String[]}
+     */
+    spriteRows;
+
+    /**
+     * `#marginFill` tracks how we will fill in the blank margins on either side of the sprite.
+     * This change is not made to the stored sprite because it can interfere with some methods.
+     * @type {{left: string, right: string}}
+     */
+    #marginFill = {left: '&nbsp;', right: '&nbsp;'};
+
+    /**
+     *  @param spriteText {String|String[]} The characters that make up the sprite.
+     *  Spaces should be non-breaking (represented by `&nbsp;`)
+     *  and lines should be break tags `<br/>` or be separate elements in an array.
+     */
+    constructor(spriteText) {
+        this.spriteRows = typeof spriteText === 'string' ? spriteText.split(br) : spriteText;
+    }
+
+    /**
+     * finds the width of the sprite
+     * @returns {{width: number, start: number, end: number}}
+     */
+    #findWidth(){
```

Comment:
Why do some properties and functions start with a #?  I'm not familiar with this convention.

Reply:
It makes them private properties/methods. I suppose it doesn't really add anything, so I can remove them if that makes them more clear.